### PR TITLE
 Implementing FeaturedArtWorkLoadingView

### PR DESCRIPTION
**Pull request by AnnaBanani**

### CODE REVIEW

File: TheMET/TheMET/FeaturedArtViewController.swift [hunk 1]
```diff
@@ -11,4 +11,13 @@ import UIKit
 
 class FeaturedArtViewController: UIViewController {
     
+    override func viewDidLoad() {
+           super.viewDidLoad()
+           
+        let newView = FeaturedArtWorkLoadingView.constructView()
+        newView.awakeFromNib()
```
Comment (Ssimboss) [inline TheMET/TheMET/FeaturedArtViewController.swift:18]: You don't need to call `awakeFromNib` on the view. It is called automatically when it is loaded from XIB

File: TheMET/TheMET/FeaturedArtViewController.swift [hunk 2]
```diff
@@ -11,4 +11,13 @@ import UIKit
...
+        self.view.addSubview(newView)
+
```
Comment (Ssimboss) [inline TheMET/TheMET/FeaturedArtViewController.swift:20]: `newView` is added to controller's view. But the constraints are not specified. Pls setup top, bottom, leading and trailing constraints of this view.

File: TheMET/TheMET/FeaturedArtWorkLoadingView/FeaturedArtWorkLoadingView.swift [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+//
+//  Created by Анна Ситникова on 18/04/2023.
+//
+
+import Foundation
+import UIKit
+
+class FeaturedArtWorkLoadingView: UIView {
+    
+    static let identifier = "FeaturedArtWorkLoadingView"
```
Comment (Ssimboss) [inline TheMET/TheMET/FeaturedArtWorkLoadingView/FeaturedArtWorkLoadingView.swift:13]: Bad naming. This is not an identifier, it is a XIB file name.

File: TheMET/TheMET/FeaturedArtWorkLoadingView/FeaturedArtWorkLoadingView.swift [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    @IBOutlet var textLabel: UILabel!
+    @IBOutlet var imageView: UIImageView!
+    
+    static func constructView() -> FeaturedArtWorkLoadingView {
+        let nib = UINib(nibName: FeaturedArtWorkLoadingView.identifier, bundle: nil)
+        return nib.instantiate(withOwner: nil).first as! FeaturedArtWorkLoadingView
+        }
+            
+    func configureImageAndText(image : UIImage, text : String) {
```
Comment (Ssimboss) [inline TheMET/TheMET/FeaturedArtWorkLoadingView/FeaturedArtWorkLoadingView.swift:24]: This method is not used

File: TheMET/TheMET/FeaturedArtWorkLoadingView/FeaturedArtWorkLoadingView.xib [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                <imageView clipsSubviews="YES" userInteractionEnabled="NO" contentMode="scaleAspectFit" horizontalHuggingPriority="251" verticalHuggingPriority="251" translatesAutoresizingMaskIntoConstraints="NO" id="6Z5-6V-4bl">
+                    <rect key="frame" x="121.66666666666669" y="209" width="150" height="150"/>
+                    <constraints>
+                        <constraint firstAttribute="width" constant="150" id="WUA-7F-DSq"/>
+                        <constraint firstAttribute="height" constant="150" id="wxW-0D-flG"/>
+                    </constraints>
+                </imageView>
+            </subviews>
+            <viewLayoutGuide key="safeArea" id="vUN-kp-3ea"/>
+            <color key="backgroundColor" red="0.082352941176470587" green="0.12156862745098039" blue="0.14901960784313725" alpha="1" colorSpace="custom" customColorSpace="sRGB"/>
```
Comment (Ssimboss) [inline TheMET/TheMET/FeaturedArtWorkLoadingView/FeaturedArtWorkLoadingView.xib:34]: I suppose the background color of this container must be `clear`.

File: TheMET/TheMET/FeaturedArtWorkLoadingView/FeaturedArtWorkLoadingView.xib [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        <placeholder placeholderIdentifier="IBFilesOwner" id="-1" userLabel="File's Owner"/>
+        <placeholder placeholderIdentifier="IBFirstResponder" id="-2" customClass="UIResponder"/>
+        <view contentMode="scaleToFill" id="iN0-l3-epB" customClass="FeaturedArtWorkLoadingView" customModule="TheMET" customModuleProvider="target">
+            <rect key="frame" x="0.0" y="0.0" width="393" height="700"/>
+            <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
+            <subviews>
+                <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Label" textAlignment="center" lineBreakMode="tailTruncation" numberOfLines="0" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="FZW-5i-T6V">
+                    <rect key="frame" x="96.666666666666686" y="508.99999999999994" width="200" height="20.333333333333314"/>
+                    <constraints>
+                        <constraint firstAttribute="width" constant="200" id="5c2-36-XCA"/>
```
Comment (Ssimboss) [inline TheMET/TheMET/FeaturedArtWorkLoadingView/FeaturedArtWorkLoadingView.xib:19]: Are you sure you want this label 200px width on all the screen sizes? I supposed the leading and trailing space should be setup instead.

**Review by Ssimboss — CHANGES_REQUESTED** [review]

File: TheMET/TheMET/FeaturedArtViewController.swift (hunk 1 above)
Comment (AnnaBanani) [reply TheMET/TheMET/FeaturedArtViewController.swift:18]: fixed

File: TheMET/TheMET/FeaturedArtViewController.swift (hunk 2 above)
Comment (AnnaBanani) [reply TheMET/TheMET/FeaturedArtViewController.swift:20]: fixed

File: TheMET/TheMET/FeaturedArtWorkLoadingView/FeaturedArtWorkLoadingView.swift (hunk 3 above)
Comment (AnnaBanani) [reply TheMET/TheMET/FeaturedArtWorkLoadingView/FeaturedArtWorkLoadingView.swift:13]: fixed

File: TheMET/TheMET/FeaturedArtWorkLoadingView/FeaturedArtWorkLoadingView.swift (hunk 4 above)
Comment (AnnaBanani) [reply TheMET/TheMET/FeaturedArtWorkLoadingView/FeaturedArtWorkLoadingView.swift:24]: fixed

File: TheMET/TheMET/FeaturedArtWorkLoadingView/FeaturedArtWorkLoadingView.xib (hunk 5 above)
Comment (AnnaBanani) [reply TheMET/TheMET/FeaturedArtWorkLoadingView/FeaturedArtWorkLoadingView.xib:34]: fixed

File: TheMET/TheMET/FeaturedArtWorkLoadingView/FeaturedArtWorkLoadingView.xib (hunk 6 above)
Comment (AnnaBanani) [reply TheMET/TheMET/FeaturedArtWorkLoadingView/FeaturedArtWorkLoadingView.xib:19]: fixed

**Review by Ssimboss — APPROVED** [review]

**Pull request merged**